### DISCUSSION TITLE
Generated album UIDs should not have a hyphen at the end of the UID

### DISCUSSION THREAD

**Discussion by jessienab** [LycheeOrg/Lychee#2551] [Ideas] [answered]
### Enhancement

Example:
photos.localhost/gallery/N4tkibPPuy2bLfsXcBRwMq10-

On many Android devices, this URL is not hyperlinked properly as the trailing hyphen is not considered part of the URL. One such case is with Signal Android:

https://github.com/signalapp/Signal-Android/issues/11310
https://github.com/signalapp/Signal-Android/issues/12288
https://github.com/signalapp/Signal-Android/issues/13049

However on Signal's side it seems to be a Won't Fix for them to not use linkifier, but ignoring that... it would probably be ideal otherwise.

Just a thought :)

**Comment by ildyria** [comment]
They are not that common (15/795 on my DB).
Feel free to propose a fix here:
https://github.com/LycheeOrg/Lychee/blob/master/app/Models/Extensions/HasRandomIDAndLegacyTimeBasedID.php#L154

**Comment by d7415** [comment]
> Example:
> photos.localhost/gallery/N4tkibPPuy2bLfsXcBRwMq10-

FWIW that URL works in Signal. The first issue you posted (the only one relating to `-`) is marked fixed.

**Reply by jessienab** [reply]
> > Example:
> > photos.localhost/gallery/N4tkibPPuy2bLfsXcBRwMq10-
> 
> FWIW that URL works in Signal. The first issue you posted (the only one relating to `-`) is marked fixed.

Just curious, is that Signal Android, iOS, or Desktop; and which version? As some friends are still experiencing the lack of hyphen being part of the hyperlink, with Signal Android. It would be helpful to further report to Signal :)

**Reply by ildyria** [reply]
Just tested on my Samsung Galaxy & Signal 7.17.6, can confirm it works as expected. :|

**Reply by d7415** [reply]
Android here too. May be able to test ios tomorrow. (Although as your issue links were all android I'd guess your friends are too?)

**Reply by jessienab** [reply]
> Just tested on my Samsung Galaxy & Signal 7.17.6, can confirm it works as expected. :|

> Android here too. May be able to test ios tomorrow. (Although as your issue links were all android I'd guess your friends are too?)

Yes, my friends device was Android... it really seems very device and software build specific then :worried: 

See my reply further down. Thank you both for testing!

**Comment by jessienab** [comment]
Hello, I opened the following PR: https://github.com/LycheeOrg/Lychee/pull/2554

based on input from an associate of mine who is a dedicated PHP developer. Since I am not, and since I have not provided a test / confirmed this doesn't break anything, I am fine if the PR/change is not accepted. Realistically this is the extent of my input on this issue, which I fully recognize is absolutely a rare and odd edge case...

It also seems per other commenters that this is VERY device specific as various other Android devices appear to not have issues parsing the URL with a hyphen at the end...

Therefore, if desired, this discussion and the PR may be closed if needed, as this could very well be all for naught.

Regardless, thank you very much all for the input and testing!

**Reply by ildyria** [reply, marked as answer]
I fixed your PR :)
Check it out.
